### PR TITLE
Perform actual refresh on `v1/datasets/:name/refresh` Api call

### PR DESCRIPTION
**Pull request by sgrebnov**
Adds logic to trigger dataset refresh on `v1/datasets/:name/refresh` Api call

This implements basic dataset refresh scenarios for M1:  https://github.com/spiceai/spiceai/issues/1057

### CODE REVIEW

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/src/datafusion.rs [hunk 1]
```diff
@@ -284,6 +293,30 @@ impl DataFusion {
         Ok(())
     }
 
+    pub async fn trigger_table_refresh(&self, dataset_name: &str) -> Result<()> {
```
Comment (lukekim) [inline crates/runtime/src/datafusion.rs]: Weird name. I'd suggest `pub async fn refresh_table(`

**Review by ewgenius — DISMISSED** [review]

File: crates/runtime/src/datafusion.rs [hunk 2]
```diff
@@ -284,6 +293,30 @@ impl DataFusion {
...
+        let table = self
+            .ctx
+            .table_provider(OwnedTableReference::bare(dataset_name.to_string()))
```
Comment (lukekim) [inline crates/runtime/src/datafusion.rs:311]: Seems like there should be some output to console like `Refresh for dataset <name> started...`

Comment (sgrebnov) [reply crates/runtime/src/datafusion.rs:311]: Confirm there is already a system message when refresh is initiated 👍
```bash
2024-04-12T02:34:01.868025Z  INFO runtime::accelerated_table: Refreshing data for recent_blocks
```

File: crates/runtime/src/datafusion.rs (hunk 1 above)
Comment (sgrebnov) [reply crates/runtime/src/datafusion.rs]: Updated - thanks!

**Review by lukekim — APPROVED** [review]

**Pull request merged**